apps/aecore/src/aec_headers.erl

```diff
@@ -100,3 +107,9 @@ deserialize_from_binary(B) ->
 hash_header(H) ->
     {ok, BinaryH} = serialize_to_binary(H),
     {ok, aec_sha256:hash(BinaryH)}.
+
+replace_empty_pow(#header{pow_evidence = no_value} = Header) ->
+    Header#header{pow_evidence = 0};
```

Comment:
I suggest to embed this conditional in `serialize_to_map` if possible, or at least do not export `replace_empty_pow` as it generates a `#header` not complying to the `#header.pow_evidence` type / default value.

Reply:
Same for block.